lua/nvim-lsp-installer/jobs/outdated-servers/github_release_file.lua

```diff
@@ -0,0 +1,29 @@
+local github = require "nvim-lsp-installer.core.clients.github"
+local CheckResult = require "nvim-lsp-installer.jobs.outdated-servers.check-result"
+
+---@param server Server
+---@param source InstallReceiptSource
+---@param on_result fun(result: CheckResult)
+return function(server, source, on_result)
+    github.fetch_latest_release(
```

Comment:
it's probably time to add `gh` integration? 😄

Reply:
Hehe I don't mind writing these HTTP client calls manually. This is however starting to feel a whole lot like early days Node.js dev. I wouldn't be surprised if core primitives for async/await, HTTP, fs, path find its way into neovim core (and then lambda expressions in LuaJIT, somehow) :D.

Also, I want to keep amount of dependencies minimal. Once (and if) the clienty stuff in lspconfig makes it to core and all what remains are server configs, I would actually even consider vendoring/bundling those inside this plugin entirely.